static/src/javascripts-legacy/bootstraps/enhanced/main.js

```diff
@@ -163,10 +163,17 @@ define([
         }
 
         // use a #force-sw hash fragment to force service worker registration for local dev
-        if ((window.location.protocol === 'https:' && config.page.section !== 'identity') || window.location.hash === '#force-sw') {
+        if ((window.location.protocol === 'https:' && config.page.section !== 'identity') || window.location.hash.indexOf('force-sw') > -1) {
             var navigator = window.navigator;
             if (navigator && navigator.serviceWorker) {
-                navigator.serviceWorker.register('/service-worker.js');
+                navigator.serviceWorker.register('/service-worker.js')
+                .then(function () {
+                    return navigator.serviceWorker.ready;
```

Comment:
Why is this inside the then of the register? Could it not be `navigation.serviceWorker.ready.then()` at the same level as the register?